bigtable-dataflow-parent/bigtable-beam-import/src/main/java/com/google/cloud/bigtable/beam/hbasesnapshots/ImportJobFromHbaseSnapshot.java

```diff
@@ -0,0 +1,133 @@
+ * and want to import snapshot gs://$HBASE_EXPORT_ROOT_PATH/.hbase-snapshot/$SNAPSHOT_NAME into
+ * Cloud Bigtable $TABLE in $INSTANCE, execute the following command to run the job directly:
+ *
+ * <pre>
+ * mvn compile exec:java \
+ *   -DmainClass=com.google.cloud.bigtable.beam.hbasesnapshots.ImportJobFromHbaseSnapshot \
+ *   -Dexec.args="--runner=DataflowRunner \
+ *                --stagingLocation=gs://$STAGING_PATH \
+ *                --project=$PROJECT \
+ *                --bigtableInstanceId=$INSTANCE \
+ *                --bigtableTableId=$TABLE \
+ *                --hbaseSnapshotSourceDir=gs://$HBASE_EXPORT_ROOT_PATH \
+ *                --snapshotName=$SNAPSHOT_NAME
+ * </pre>
+ */
+@InternalExtensionOnly
+public class ImportJobFromHbaseSnapshot {
+  private static final Log LOG = LogFactory.getLog(ImportJobFromHbaseSnapshot.class);
+
+  public interface ImportOptions extends ImportJob.ImportOptions {
+    @Description("The HBase root dir where HBase snapshot files resides.")
+    String getHbaseSnapshotSourceDir();
+
+    @SuppressWarnings("unused")
+    void setHbaseSnapshotSourceDir(String hbaseSnapshotSourceDir);
+
+    @Description("Snapshot name")
+    String getSnapshotName();
+
+    @SuppressWarnings("unused")
+    void setSnapshotName(String snapshotName);
+  }
+
+  public static void main(String[] args) throws Exception {
+    PipelineOptionsFactory.register(ImportOptions.class);
+
+    ImportOptions opts =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(ImportOptions.class);
+
+    LOG.info("Building Pipeline");
+    Pipeline pipeline = buildPipeline(opts);
+    LOG.info("Running Pipeline");
+    PipelineResult result = pipeline.run();
+
+    if (opts.getWait()) {
+      Utils.waitForPipelineToFinish(result);
+    }
+  }
+
+  @VisibleForTesting
+  static Pipeline buildPipeline(ImportOptions opts) throws Exception {
+
+    Pipeline pipeline = Pipeline.create(Utils.tweakOptions(opts));
+    Configuration configuration =
+        new HBaseSnapshotInputConfigBuilder()
+            .setProjectId(opts.getProject())
+            .setHbaseSnapshotSourceDir(opts.getHbaseSnapshotSourceDir())
+            .setSnapshotName(opts.getSnapshotName())
+            .setRestoreDirSuffix(opts.getJobName())
+            .build();
+    PCollection<KV<ImmutableBytesWritable, Result>> readResult =
+        pipeline.apply(
+            "Read from HBase Snapshot",
+            HadoopFormatIO.<ImmutableBytesWritable, Result>read().withConfiguration(configuration));
+
+    readResult
+        .apply("Create Mutations", ParDo.of(new HBaseResultToMutationFn()))
+        .apply(
+            "Write to Bigtable",
+            CloudBigtableIO.writeToTable(TemplateUtils.BuildImportConfig(opts)));
+
+    final List<KV<String, String>> tempFiles =
+        Arrays.asList(
+            KV.of(
+                opts.getHbaseSnapshotSourceDir(),
```

Comment:
No, the source file will not be deleted, the hbaseSnapshoSourceDir is used to get the bucket name(which the temp folder reside in). 

Actually the reason I passed `hbaseSnapshotSourceDir` in instead of just `GcsPath.fromUri(hbaseSnapshotDir).getBucket()` is because we need to make sure restoreDir does not include hbaseSnapshotSourceDir and I believe that is some check belongs to the DoFn itself instead of in the buildPipeline() call.

To avoid confustion, I renamed the variable from `tempFiles` to `sourceAndRestoreFolders`, hopefully together with the name `CleanupHBaseSnapshotRestoreFilesFn`, it will cause less confusion.

As for the failure handling, I can see 2 options:
1. ask the user to fix the pipeline and launch a replacement pipeline(so the same restore folder is used and will be cleanup)
2. ask the user to manually delete the restore folder

I am not sure there is a way to guarantee cleanup will be executed if the pipeline fails, since the restore folder will be generated even start actual reading. in my experiment with customized worker images, sometimes the job will be killed before even start reading from snapshots, so in that case there is nothing the pipeline can do to clean it up.

Reply:
Ah, sorry, I misread the code I thought you were making a list of that include the source dir. 

I dont think you need to figure out how to cleanup on failure right now. Just add it to the docs and we can figure it out later

Reply:
Done.

I added a paragraph in the header comments for `ImportJobFromHbaseSnapshot`, eventually when all pieces are together, we would like to also provide a README.md for this job. But I would avoid that right now since the validation part isn't checked in yet.